cache/memcached_client.go

```diff
@@ -564,3 +597,114 @@ func (c *MemcachedClient) resolveAddrs() error {
 
 	return c.selector.SetServers(servers...)
 }
+
+func (c *MemcachedClient) CompareAndSwap(ctx context.Context, key string, value []byte, ttl time.Duration) error {
+	item := &memcache.Item{
+		Key:        key,
+		Value:      value,
+		Expiration: int32(ttl.Seconds()),
+	}
+	return c.compareAndSwap(ctx, key, value, ttl, func(ctx context.Context, key string, value []byte, ttl time.Duration) error {
+		var err error
+		select {
+		case <-ctx.Done():
+			err = ctx.Err()
+		default:
+			err = c.client.CompareAndSwap(item)
+		}
+		return err
+	})
+}
+
+func (c *MemcachedClient) FlushAll(ctx context.Context) error {
+	return c.flushAll(ctx, func(ctx context.Context) error {
+		var err error
+		select {
+		case <-ctx.Done():
+			err = ctx.Err()
+		default:
+			err = c.client.FlushAll()
+		}
+		return err
+	})
+}
+
+func (c *MemcachedClient) touch(ctx context.Context, key string, ttl time.Duration, f func(ctx context.Context, key string, ttl time.Duration) error) error {
+	start := time.Now()
+	c.metrics.operations.WithLabelValues(opTouch).Inc()
+
+	err := f(ctx, key, ttl)
+	if err != nil {
+		level.Debug(c.logger).Log(
+			"msg", "failed to touch cache item",
+			"key", key,
+			"err", err,
+		)
+		c.trackError(opTouch, err)
+	} else {
+		c.metrics.duration.WithLabelValues(opTouch).Observe(time.Since(start).Seconds())
+	}
+
+	return err
+}
+
+func (c *MemcachedClient) increment(ctx context.Context, key string, delta uint64, f func(ctx context.Context, key string, delta uint64) (uint64, error)) (uint64, error) {
```

Comment:
Thank you but now there's not really any reason to keep these methods around. Instead, they could be combined with their corresponding public methods.  WDYT?

Reply:
I prefer atomized functions, but in this case, the function stays small, so this is okay.